hive/src/test/java/com/netflix/iceberg/hive/HiveTableBaseTest.java

```diff
@@ -0,0 +1,229 @@
+  }
+
+  private String getDerbyPath() {
+    return new File(hiveLocalDir, "metastore_db").getPath();
+  }
+
+  private final TServer thriftServer() throws IOException,
+          TTransportException,
+          MetaException,
+          InvocationTargetException,
+          NoSuchMethodException,
+          IllegalAccessException,
+          NoSuchFieldException {
+    final TServerSocketKeepAlive socket = new TServerSocketKeepAlive(new TServerSocket(0));
+    this.hiveConf = hiveConf(new Configuration(), socket.getServerSocket().getLocalPort());
+    HiveMetaStore.HMSHandler baseHandler = new HiveMetaStore.HMSHandler("new db based metaserver", hiveConf);
+    IHMSHandler handler = RetryingHMSHandler.getProxy(hiveConf, baseHandler, true);
+    final TTransportFactory transportFactory = new TTransportFactory();
+    final TSetIpAddressProcessor<IHMSHandler> processor = new TSetIpAddressProcessor<>(handler);
+
+    TThreadPoolServer.Args args = new TThreadPoolServer.Args(socket)
+            .processor(processor)
+            .transportFactory(transportFactory)
+            .protocolFactory(new TBinaryProtocol.Factory())
+            .minWorkerThreads(3)
+            .maxWorkerThreads(5);
+
+    return new TThreadPoolServer(args);
+  }
+
+  private final void runScript(String dbURL) throws SQLException, IOException {
+    Connection connection = DriverManager.getConnection(dbURL);
+    ScriptRunner scriptRunner = new ScriptRunner(connection, true, true);
+
+    final URL hiveSqlScript = getClass().getClassLoader().getResource("hive-txn-schema-2.3.0.derby.sql");
+    Reader reader = new BufferedReader(new FileReader(new File(hiveSqlScript.getFile())));
+    scriptRunner.runScript(reader);
+  }
+
+  private void createIfNotExistsCatalog(String catalogName) throws TException {
+    try {
+      metastoreClient.getCatalog(catalogName);
+    } catch(NoSuchObjectException e) {
+      String catalogPath = Paths.get(hiveLocalDir.getAbsolutePath(), catalogName + ".catalog").toString();
+      metastoreClient.createCatalog(new Catalog(catalogName, catalogPath));
+    }
+  }
+  
+  protected String getDBPath() {
+   return Paths.get(hiveLocalDir.getAbsolutePath(), DB_NAME + ".db").toAbsolutePath().toString();
+  }
+
+  protected String getTableBasePath(String tableName) {
+    return Paths.get(getDBPath(), tableName).toAbsolutePath().toString();
+  }
+
+  protected String getTableLocation(String tableName) {
+    return new Path("file", null, Paths.get(getTableBasePath(tableName), "empty").toString()).toString();
+  }
+
+  protected String metadataLocation(String tableName) {
+    return Paths.get(getTableBasePath(tableName), "metadata").toString();
+  }
+
+  protected List<String> metadataFiles(String tableName) {
+    return Arrays.stream(new File(metadataLocation(tableName)).listFiles())
+            .map(f -> f.getAbsolutePath())
+            .collect(Collectors.toList());
+  }
+
+  protected List<String> snapShotFiles(String tableName) {
```

Comment:
It's confusing to call these snapshot files because they are table metadata files that include several snapshots. What about using metadataVersionFiles to avoid a conflict with metadataFiles?